### PR TITLE
fix: Increase unit test coverage (string_schema, format_responses)

### PR DESCRIPTION
**Pull request by timmc-edx**
This includes a small change to logging to make it easier to test.

**Merge checklist:**
- [ ] Version bumped
- [ ] Changelog record added
- [x] Documentation updated (not only docstrings)
- [x] Commits are squashed

### CODE REVIEW

**Comment by timmc-edx** [comment]
I set out to improve coverage, but it turns out most of the missing coverage is in https://github.com/openedx/openedx-events/blob/main/openedx_events/tests/utils.py -- which is all test utilities that maybe we don't care about coverage for. :-) I'd be happy to add a no-cover pragma if you think it makes sense.

File: openedx_events/tests/test_tooling.py [hunk 1]
```diff
@@ -37,6 +53,14 @@ def setUp(self):
             data=self.data_attr,
         )
 
+        self.receiver_error = Exception("fake error")
+
+        def error_receiver(*args, **kwargs):
+            raise self.receiver_error
+
+        self.ok_receiver = Mock(return_value="success")
+        self.error_receiver = error_receiver
```
Comment (timmc-edx) [inline openedx_events/tests/test_tooling.py:62]: Author note: These test receivers and `receivers_attached` could be used in other tests in this file, which is why they're up here. If you don't think we'll get around to using them in other tests, I can collapse them back down into the one place they're used.

**Comment by timmc-edx** [comment]
I also noticed that `InstantiationError` is no longer referenced, having previously been removed in #23. I don't see references elsewhere in edx or openedx. Can probably remove safely. (It's responsible for a few lines of coverage loss.)

File: openedx_events/tests/test_tooling.py (hunk 1 above)
Comment (mariajgrimaldi) [reply openedx_events/tests/test_tooling.py:62]: This looks really cool! And I see it can be used elsewhere if needed

**Review by mariajgrimaldi — APPROVED** [review]

**Pull request merged**